pkg/component/component.go

```diff
@@ -564,9 +565,10 @@ func ensureAndLogProperResourceUsage(resourceMin, resourceMax *string, resourceN
 //	componentConfig: Component configuration
 //	envSpecificInfo: Component environment specific information, available if uses devfile
 //  cmpExist: true if components exists in the cluster
+//  endpointMap: value is devfile endpoint entry, key is the TargetPort for each enpoint entry
 // Returns:
 //	err: Errors if any else nil
-func ApplyConfig(client *occlient.Client, kClient *kclient.Client, componentConfig config.LocalConfigInfo, envSpecificInfo envinfo.EnvSpecificInfo, stdout io.Writer, cmpExist bool) (err error) {
+func ApplyConfig(client *occlient.Client, kClient *kclient.Client, componentConfig config.LocalConfigInfo, envSpecificInfo envinfo.EnvSpecificInfo, stdout io.Writer, cmpExist bool, endpointMap map[int32]parsercommon.Endpoint) (err error) {
```

Comment:
So I've ran into issues before with regards to using maps as a parameter before as it's usually not in the correct order. 

Can you pass in an array of structs or something instead? Endpoint's would be different each time.

Reply:
@cdrage Why would this be an issue? The order should not matter here.